transport/httptransport/http_transport.go

```diff
@@ -0,0 +1,241 @@
+package httptransport
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+
+	"github.com/filecoin-project/boost/transport/types/transferstatus"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/libp2p/go-eventbus"
+	"github.com/libp2p/go-libp2p-core/event"
+
+	"github.com/filecoin-project/boost/transport/httptransport/pb"
+	"github.com/filecoin-project/boost/transport/types"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/filecoin-project/boost/transport"
+)
+
+var log = logging.Logger("http-transport")
+
+const (
+	readBufferSize = 4096
```

Comment:
For efficiency we may want to make `readBufferSize` bigger than 4096. I'm not sure what the standard is here though.

Reply:
It is now 64K. Let's iterate if needed later.